src/community/script/core/src/main/java/org/geoserver/script/function/ScriptFunctionFactory.java

```diff
@@ -100,40 +98,34 @@ ScriptFunction function(Name name) {
             synchronized(this) {
                 function = functions.get(name);
                 if (function == null) {
-                    try {
-                        ScriptManager scriptMgr = scriptMgr();
+                    ScriptManager scriptMgr = scriptMgr();
 
-                        File filterRoot = scriptMgr.getFunctionRoot();
-                        File f = null;
-                        if (name.getNamespaceURI() != null) {
-                            f = new File(filterRoot, name.getLocalPart()+"."+name.getNamespaceURI());
-                        }
-                        else {
-                            //look for a file based on basename
-                            for (String filename : filterRoot.list()) {
-                                if (name.getLocalPart().equals(getBaseName(filename))) {
-                                    f = new File(filterRoot, filename);
-                                    break;
-                                }
+                    Resource filterRoot = scriptMgr.function();
+                    Resource f = null;
+                    if (name.getNamespaceURI() != null) {
+                        f = filterRoot.get(name.getLocalPart()+"."+name.getNamespaceURI());
```

Comment:
Need spaces around operators. @NielsCharlier  I know that the bad formatting was there originally but feel free to fix as you go. :-)